inst/include/dust/gpu/rng.hpp

```diff
@@ -0,0 +1,144 @@
+#ifndef DUST_GPU_RNG_HPP
+#define DUST_GPU_RNG_HPP
+
+#include <dust/gpu/xoshiro.hpp>
+#include <dust/gpu/distr/binomial.hpp>
+#include <dust/gpu/distr/normal.hpp>
+#include <dust/gpu/distr/poisson.hpp>
+#include <dust/gpu/distr/uniform.hpp>
+
+namespace dust {
+
+struct RNGptr {
+  uint64_t* state_ptr;
+  unsigned int state_stride;
+  unsigned int particle_stride;
+};
+
+// Read state from global memory
+template <typename T>
+__device__
+rng_state_t<T> loadRNG(RNGptr& rng_state, int p_idx) {
```

Comment:
I would suggest changing the name of the `RNGptr` argument in this function to `global_memory_rng_state` or `d_rng_state` to make `rng_state` less overloaded, and be clearer about where in memory it resides

Reply:
257aa11